distributed/scheduler.py

```diff
@@ -2324,7 +2324,10 @@ def decide_worker(self, ts: TaskState) -> WorkerState:
         if ts._dependencies or valid_workers is not None:
             ws = decide_worker(
                 ts,
-                self._workers_dv.values(),
+                self._workers_dv.values() if compiled else dict.values(self._workers),
+                self._idle_dv.values() if compiled else dict.values(self._workers),
+                # ^ NOTE: For performance, these must be actual `dict_values`, not `SortedDictValues`.
+                # In Cython, `_workers_dv` is a plain dict, but in plain Python, it's still a `SortedDict`.
```

Comment:
Is it bad to always coerce? How much does that cost us?

Reply:
I suppose they're actually exactly equivalent statements, so I think it costs us nothing (unless Cython interprets it differently?). We could just do `dict.values(self._workers_dv)`. Just a bit hacky. Feels more like a note to self that if we ever had Cython be the default / distributed compiled wheels, we could maybe remove the conditional.

Reply:
Yeah I don't see a real alternative. Cython can `cast` whereas Python can't. So we are just handling that inherent awkwardness

I suppose we could just use an ordinary `dict` instead of a `SortedDict` if that were acceptable